plugins/woocommerce/src/Blocks/BlockTypes/AddToCartWithOptions.php

```diff
@@ -208,4 +215,106 @@ public function add_to_cart_redirect_filter( $url ) {
 
 		return $url;
 	}
+
+	/**
+	 * Add a core/button variation for the Add To Cart Button block.
+	 * It registers a new block `withRole` attribute
+	 * to identify when it's active.
+	 *
+	 * @param array  $args Block type args.
+	 * @param string $block_name Block name.
+	 * @return array
+	 */
+	public function add_core_button_variation_for_add_to_cart_button( $args, $block_name ) {
+		if ( 'core/button' !== $block_name ) {
+			return $args;
+		}
+
+		/*
+		 * Add withRole attribute to the block.
+		 * This attribute is used to identify when the block is active.
+		 */
+		$args['attributes']['withRole'] = array(
+			'type' => 'string',
+		);
+
+		if ( ! isset( $args['variations'] ) ) {
+			$args['variations'] = array();
+		}
+
+		$args['variations'][] = array(
+			'name'        => 'woocommerce/product-add-to-cart-with-options-button',
```

Comment:
Should we remove `product-` here for consistency?

```suggestion
			'name'        => 'woocommerce/add-to-cart-with-options-button',
```